cfme/containers/provider/openshift.py

```diff
@@ -384,3 +385,30 @@ def sync_ssl_certificate(self):
             self.appliance.restart_evm_service()
             self.appliance.wait_for_evm_service()
             self.appliance.wait_for_web_ui()
+
+    @variable(alias='rest')
+    def is_refreshed(self, refresh_timer=None, refresh_delta=600):
+        """ Overrides cfme.common.provider.BaseProvider#is_refreshed.
+            Add validate_status check after provider creation,
+            In order to make sure that openshift provider data refresh completed
+                Args:
+                    refresh_timer : wait_for.RefreshTimer Object or None
+                    refresh_delta : refresh delta between times on second
+                Returns:
+                     Boolean or TimedOutError exception in case data referesh didn't completed
+        """
+        valid = False
+        # check refresh time
+        if not super(OpenshiftProvider, self).is_refreshed():
+            return valid
+
+        # check the detail page matches the Providers information
+        try:
+            self.validate_stats(ui=False)
+            valid = True
+        except TimedOutError:
```

Comment:
__QUESTION__
I think you can remove the `except` because it's already raise this TimeOutError if it fails. Or you just wanted more informative exception?